sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosItemRequestOptions.java

```diff
@@ -101,17 +101,21 @@ public CosmosItemRequestOptions setIfNoneMatchETag(String ifNoneMatchETag) {
      * @return the consistency level.
      */
 
-    ConsistencyLevel getConsistencyLevel() {
+    public ConsistencyLevel getConsistencyLevel() {
         return consistencyLevel;
     }
 
     /**
-     * Sets the consistency level required for the request.
+     * Sets the consistency level required for the request. The effective consistency level
+     * can only be reduce for read/query requests. So when the Account's default consistency level
+     * is for example Session you can specify on a request-by-request level for individual requests
+     * that Eventual consistency is sufficient - which could reduce the latency and RU charges for this
+     * request but will not guarantee session consistency (read-your-own-write) anymore
```

Comment:
If the account default consistency is session, client default consistency is set to be eventual, but for an individual request consistency is set to session, we will have a problem.

By default in java v4 we don't capture session tokens for non session consistency.
we should document that if the user wants to use this per request option, the user has to enable per request session capturing on the cosmos-client-builder and we should make the method for that public.

see `CosmosClientBuilder:sessionCapturingOverrideEnabled()`

`CosmosItemRequestOption.setConsistencyLevel()` and `CosmosClientBuilder:sessionCapturingOverrideEnabled()` should go hand in hand.

Reply:
Nice catch - thanks. Added documentation and made sessionCapturingOverrideEnabled public